From c0313059e017f4fc60cbe9fcceb689e3d83c0b2f Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Mon, 16 Oct 2023 12:20:54 +0200
Subject: [PATCH 0009/1695] core: fix header file guard names

Fixes header file guards macro name prefix (mostly missing occurrences)
that should reflect the file base directory sub-path where applicable.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 9bb1aa102ffd5a2e22345e7ec715e0ed2a94532b core: fix header file guard names]
Change-Id: I64ca8f3e4364b1f3f5a5e5b00a8d1b0e6113eaed
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
[etienne: cherry-picked from fbe66cf83199aa6a2aca9f93384cf1ad9185a5f6]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/arch/arm/include/crypto/ghash-ce-core.h       | 6 +++---
 core/arch/arm/include/kernel/tee_l2cc_mutex.h      | 7 ++++---
 core/arch/arm/include/kernel/tlb_helpers.h         | 6 +++---
 core/arch/arm/include/kernel/tz_ssvce_def.h        | 6 +++---
 core/arch/arm/include/kernel/tz_ssvce_pl310.h      | 6 +++---
 core/arch/arm/include/mm/core_mmu_arch.h           | 6 +++---
 core/arch/arm/include/scmi/scmi_server.h           | 6 +++---
 core/arch/arm/include/sm/optee_smc.h               | 6 +++---
 core/arch/arm/include/sm/std_smc.h                 | 4 ++--
 core/arch/arm/include/sm/teesmc_opteed.h           | 6 +++---
 core/arch/arm/include/sm/teesmc_opteed_macros.h    | 6 +++---
 core/arch/riscv/include/kernel/tee_l2cc_mutex.h    | 6 +++---
 core/arch/riscv/include/kernel/tlb_helpers.h       | 6 +++---
 core/arch/riscv/include/mm/core_mmu_arch.h         | 6 +++---
 core/arch/riscv/include/tee/optee_abi.h            | 6 +++---
 core/arch/riscv/include/tee/teeabi_opteed.h        | 6 +++---
 core/arch/riscv/include/tee/teeabi_opteed_macros.h | 6 +++---
 core/include/crypto/crypto_se.h                    | 4 ++--
 core/include/drivers/amlogic_uart.h                | 6 +++---
 core/include/drivers/atmel_uart.h                  | 6 +++---
 core/include/drivers/bcm/bnxt.h                    | 4 ++--
 core/include/drivers/bcm_gpio.h                    | 6 +++---
 core/include/drivers/bcm_hwrng.h                   | 6 +++---
 core/include/drivers/bcm_sotp.h                    | 4 ++--
 core/include/drivers/caam_extension.h              | 2 +-
 core/include/drivers/cdns_uart.h                   | 6 +++---
 core/include/drivers/hi16xx_uart.h                 | 6 +++---
 core/include/drivers/imx/dcp.h                     | 6 +++---
 core/include/drivers/imx_ocotp.h                   | 6 +++---
 core/include/drivers/imx_uart.h                    | 6 +++---
 core/include/drivers/imx_wdog.h                    | 4 ++--
 core/include/drivers/lpc_uart.h                    | 6 +++---
 core/include/drivers/mvebu_uart.h                  | 6 +++---
 core/include/drivers/ns16550.h                     | 6 +++---
 core/include/drivers/pl011.h                       | 6 +++---
 core/include/drivers/pl022_spi.h                   | 6 +++---
 core/include/drivers/pl061_gpio.h                  | 6 +++---
 core/include/drivers/sam/at91_ddr.h                | 4 ++--
 core/include/drivers/scif.h                        | 6 +++---
 core/include/drivers/scmi-msg.h                    | 6 +++---
 core/include/drivers/scmi.h                        | 6 +++---
 core/include/drivers/serial8250_uart.h             | 6 +++---
 core/include/drivers/sp805_wdt.h                   | 6 +++---
 core/include/drivers/sprd_uart.h                   | 6 +++---
 core/include/drivers/stih_asc.h                    | 6 +++---
 core/include/drivers/stm32_bsec.h                  | 6 +++---
 core/include/drivers/stm32_etzpc.h                 | 6 +++---
 core/include/drivers/stm32mp_dt_bindings.h         | 6 +++---
 core/include/drivers/stpmic1.h                     | 6 +++---
 core/include/kernel/chip_services.h                | 4 ++--
 core/include/kernel/dt_driver.h                    | 7 +++----
 core/include/kernel/rpc_io_i2c.h                   | 6 +++---
 core/include/kernel/tee_common.h                   | 6 +++---
 core/include/kernel/tee_common_otp.h               | 6 +++---
 core/include/kernel/tee_misc.h                     | 6 +++---
 core/include/kernel/tee_ta_manager.h               | 4 ++--
 core/include/kernel/tee_time.h                     | 4 ++--
 core/include/kernel/timer.h                        | 6 +++---
 core/include/kernel/trace_ta.h                     | 6 +++---
 core/include/mm/core_memprot.h                     | 6 +++---
 core/include/mm/core_mmu.h                         | 6 +++---
 core/include/mm/tee_mm.h                           | 4 ++--
 core/include/mm/tee_mmu_types.h                    | 4 ++--
 core/include/mm/vm.h                               | 6 +++---
 core/include/ta_pub_key.h                          | 6 +++---
 core/include/tee/svc_cache.h                       | 6 +++---
 core/include/tee/tee_cryp_concat_kdf.h             | 6 +++---
 core/include/tee/tee_cryp_hkdf.h                   | 6 +++---
 core/include/tee/tee_cryp_pbkdf2.h                 | 6 +++---
 core/include/tee/tee_cryp_utl.h                    | 4 ++--
 core/include/tee/tee_fs.h                          | 6 +++---
 core/include/tee/tee_fs_key_manager.h              | 4 ++--
 core/include/tee/tee_fs_rpc.h                      | 6 +++---
 core/include/tee/tee_obj.h                         | 4 ++--
 core/include/tee/tee_pobj.h                        | 4 ++--
 core/include/tee/tee_supp_plugin_rpc.h             | 6 +++---
 core/include/tee/tee_svc.h                         | 6 +++---
 core/include/tee/tee_svc_cryp.h                    | 6 +++---
 core/include/tee/tee_svc_storage.h                 | 6 +++---
 core/include/tee/tee_ta_enc_manager.h              | 4 ++--
 80 files changed, 223 insertions(+), 223 deletions(-)

diff --git a/core/arch/arm/include/crypto/ghash-ce-core.h b/core/arch/arm/include/crypto/ghash-ce-core.h
index b0f9f5457..e2be237f8 100644
--- a/core/arch/arm/include/crypto/ghash-ce-core.h
+++ b/core/arch/arm/include/crypto/ghash-ce-core.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2017-2020, Linaro Limited
  */
 
-#ifndef __GHASH_CE_CORE_H
-#define __GHASH_CE_CORE_H
+#ifndef __CRYPTO_GHASH_CE_CORE_H
+#define __CRYPTO_GHASH_CE_CORE_H
 
 #include <inttypes.h>
 
@@ -40,4 +40,4 @@ uint32_t pmull_gcm_aes_sub(uint32_t input);
 
 void pmull_gcm_encrypt_block(uint8_t dst[], const uint8_t src[], int rounds);
 
-#endif /*__GHASH_CE_CORE_H*/
+#endif /*__CRYPTO_GHASH_CE_CORE_H*/
diff --git a/core/arch/arm/include/kernel/tee_l2cc_mutex.h b/core/arch/arm/include/kernel/tee_l2cc_mutex.h
index 1011fced9..7fb5af3e2 100644
--- a/core/arch/arm/include/kernel/tee_l2cc_mutex.h
+++ b/core/arch/arm/include/kernel/tee_l2cc_mutex.h
@@ -2,8 +2,9 @@
 /*
  * Copyright (c) 2014, STMicroelectronics International N.V.
  */
-#ifndef TEE_L2CC_MUTEX_H
-#define TEE_L2CC_MUTEX_H
+#ifndef __KERNEL_TEE_L2CC_MUTEX_H
+#define __KERNEL_TEE_L2CC_MUTEX_H
+
 #include <inttypes.h>
 #include <tee_api_types.h>
 #include <tee_api_defines.h>
@@ -47,4 +48,4 @@ static inline TEE_Result tee_set_l2cc_mutex(paddr_t *mutex __unused)
 }
 #endif
 
-#endif /* TEE_L2CC_MUTEX_H */
+#endif /* __KERNEL_TEE_L2CC_MUTEX_H */
diff --git a/core/arch/arm/include/kernel/tlb_helpers.h b/core/arch/arm/include/kernel/tlb_helpers.h
index 36a9af396..92d596aee 100644
--- a/core/arch/arm/include/kernel/tlb_helpers.h
+++ b/core/arch/arm/include/kernel/tlb_helpers.h
@@ -4,8 +4,8 @@
  * Copyright (c) 2017, Linaro Limited
  */
 
-#ifndef TLB_HELPERS_H
-#define TLB_HELPERS_H
+#ifndef __KERNEL_TLB_HELPERS_H
+#define __KERNEL_TLB_HELPERS_H
 
 #include <arm.h>
 
@@ -48,4 +48,4 @@ static inline void tlbi_va_asid(vaddr_t va, uint32_t asid)
 }
 #endif /*!__ASSEMBLER__*/
 
-#endif /* TLB_HELPERS_H */
+#endif /* __KERNEL_TLB_HELPERS_H */
diff --git a/core/arch/arm/include/kernel/tz_ssvce_def.h b/core/arch/arm/include/kernel/tz_ssvce_def.h
index 25215b537..e69e81b7a 100644
--- a/core/arch/arm/include/kernel/tz_ssvce_def.h
+++ b/core/arch/arm/include/kernel/tz_ssvce_def.h
@@ -2,8 +2,8 @@
 /*
  * Copyright (c) 2014, STMicroelectronics International N.V.
  */
-#ifndef TZ_SSVCE_DEF_H
-#define TZ_SSVCE_DEF_H
+#ifndef __KERNEL_TZ_SSVCE_DEF_H
+#define __KERNEL_TZ_SSVCE_DEF_H
 
 #include <stdint.h>
 #include <util.h>
@@ -125,4 +125,4 @@
  */
 #define CORE_ICC_ICCPMR	U(0x0004)
 
-#endif /* TZ_SSVCE_DEF_H */
+#endif /* __KERNEL_TZ_SSVCE_DEF_H */
diff --git a/core/arch/arm/include/kernel/tz_ssvce_pl310.h b/core/arch/arm/include/kernel/tz_ssvce_pl310.h
index 2bcd26ebe..1e7685d40 100644
--- a/core/arch/arm/include/kernel/tz_ssvce_pl310.h
+++ b/core/arch/arm/include/kernel/tz_ssvce_pl310.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2014, STMicroelectronics International N.V.
  */
 
-#ifndef TZ_SSVCE_PL310_H
-#define TZ_SSVCE_PL310_H
+#ifndef __KERNEL_TZ_SSVCE_PL310_H
+#define __KERNEL_TZ_SSVCE_PL310_H
 
 #include <util.h>
 #include <kernel/tz_ssvce_def.h>
@@ -21,4 +21,4 @@ void arm_cl2_cleanbypa(vaddr_t pl310_base, paddr_t start, paddr_t end);
 void arm_cl2_invbypa(vaddr_t pl310_base, paddr_t start, paddr_t end);
 void arm_cl2_cleaninvbypa(vaddr_t pl310_base, paddr_t start, paddr_t end);
 
-#endif /* TZ_SSVCE_PL310_H */
+#endif /* __KERNEL_TZ_SSVCE_PL310_H */
diff --git a/core/arch/arm/include/mm/core_mmu_arch.h b/core/arch/arm/include/mm/core_mmu_arch.h
index 42f382fd5..6c68a4578 100644
--- a/core/arch/arm/include/mm/core_mmu_arch.h
+++ b/core/arch/arm/include/mm/core_mmu_arch.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2016, Linaro Limited
  * Copyright (c) 2014, STMicroelectronics International N.V.
  */
-#ifndef __CORE_MMU_ARCH_H
-#define __CORE_MMU_ARCH_H
+#ifndef __MM_CORE_MMU_ARCH_H
+#define __MM_CORE_MMU_ARCH_H
 
 #ifndef __ASSEMBLER__
 #include <arm.h>
@@ -223,4 +223,4 @@ static inline bool core_mmu_level_in_range(unsigned int level)
 
 #endif /*__ASSEMBLER__*/
 
-#endif /* CORE_MMU_H */
+#endif /* __MM_CORE_MMU_ARCH_H */
diff --git a/core/arch/arm/include/scmi/scmi_server.h b/core/arch/arm/include/scmi/scmi_server.h
index a28fd667f..d969e238d 100644
--- a/core/arch/arm/include/scmi/scmi_server.h
+++ b/core/arch/arm/include/scmi/scmi_server.h
@@ -2,8 +2,8 @@
 /*
  * Copyright (c) 2019-2022, Linaro Limited
  */
-#ifndef SCMI_SERVER_H
-#define SCMI_SERVER_H
+#ifndef __SCMI_SCMI_SERVER_H
+#define __SCMI_SCMI_SERVER_H
 
 #include <tee_api_types.h>
 #include <types_ext.h>
@@ -70,4 +70,4 @@ static inline int scmi_server_get_channels_count(void)
 	return 0;
 }
 #endif /* CFG_SCMI_SCPFW */
-#endif /* SCMI_SERVER_H */
+#endif /* __SCMI_SCMI_SERVER_H */
diff --git a/core/arch/arm/include/sm/optee_smc.h b/core/arch/arm/include/sm/optee_smc.h
index 04a3839a2..fe9aff0e1 100644
--- a/core/arch/arm/include/sm/optee_smc.h
+++ b/core/arch/arm/include/sm/optee_smc.h
@@ -2,8 +2,8 @@
 /*
  * Copyright (c) 2015-2021, Linaro Limited
  */
-#ifndef OPTEE_SMC_H
-#define OPTEE_SMC_H
+#ifndef __SM_OPTEE_SMC_H
+#define __SM_OPTEE_SMC_H
 
 #include <stdint.h>
 
@@ -705,4 +705,4 @@
 	((((ret) & OPTEE_SMC_RETURN_RPC_PREFIX_MASK) == \
 		OPTEE_SMC_RETURN_RPC_PREFIX)))
 
-#endif /* OPTEE_SMC_H */
+#endif /* __SM_OPTEE_SMC_H */
diff --git a/core/arch/arm/include/sm/std_smc.h b/core/arch/arm/include/sm/std_smc.h
index 8c55ec72c..80e71a831 100644
--- a/core/arch/arm/include/sm/std_smc.h
+++ b/core/arch/arm/include/sm/std_smc.h
@@ -2,8 +2,8 @@
 /*
  * Copyright (c) 2017-2023, Linaro Limited
  */
-#ifndef __STD_SMC_H__
-#define __STD_SMC_H__
+#ifndef __SM_STD_SMC_H__
+#define __SM_STD_SMC_H__
 
 #include <sm/sm.h>
 
diff --git a/core/arch/arm/include/sm/teesmc_opteed.h b/core/arch/arm/include/sm/teesmc_opteed.h
index 41e7204d7..7213531aa 100644
--- a/core/arch/arm/include/sm/teesmc_opteed.h
+++ b/core/arch/arm/include/sm/teesmc_opteed.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2014, Linaro Limited
  */
 
-#ifndef TEESMC_OPTEED_H
-#define TEESMC_OPTEED_H
+#ifndef __SM_TEESMC_OPTEED_H
+#define __SM_TEESMC_OPTEED_H
 
 /*
  * This file specify SMC function IDs used when returning from TEE to the
@@ -117,4 +117,4 @@
 #define TEESMC_OPTEED_RETURN_SYSTEM_RESET_DONE \
 	TEESMC_OPTEED_RV(TEESMC_OPTEED_FUNCID_RETURN_SYSTEM_RESET_DONE)
 
-#endif /*TEESMC_OPTEED_H*/
+#endif /*__SM_TEESMC_OPTEED_H*/
diff --git a/core/arch/arm/include/sm/teesmc_opteed_macros.h b/core/arch/arm/include/sm/teesmc_opteed_macros.h
index fcc84b72f..2171c6283 100644
--- a/core/arch/arm/include/sm/teesmc_opteed_macros.h
+++ b/core/arch/arm/include/sm/teesmc_opteed_macros.h
@@ -3,11 +3,11 @@
  * Copyright (c) 2014, Linaro Limited
  */
 
-#ifndef TEESMC_OPTEED_MACROS_H
-#define TEESMC_OPTEED_MACROS_H
+#ifndef __SM_TEESMC_OPTEED_MACROS_H
+#define __SM_TEESMC_OPTEED_MACROS_H
 
 #define TEESMC_OPTEED_RV(func_num) \
 	OPTEE_SMC_CALL_VAL(OPTEE_SMC_32, OPTEE_SMC_FAST_CALL, \
 			   OPTEE_SMC_OWNER_TRUSTED_OS_OPTEED, (func_num))
 
-#endif /*TEESMC_OPTEED_MACROS_H*/
+#endif /*__SM_TEESMC_OPTEED_MACROS_H*/
diff --git a/core/arch/riscv/include/kernel/tee_l2cc_mutex.h b/core/arch/riscv/include/kernel/tee_l2cc_mutex.h
index 12b914780..9dc873a8e 100644
--- a/core/arch/riscv/include/kernel/tee_l2cc_mutex.h
+++ b/core/arch/riscv/include/kernel/tee_l2cc_mutex.h
@@ -2,7 +2,7 @@
 /*
  * Copyright 2022 NXP
  */
-#ifndef TEE_L2CC_MUTEX_H
-#define TEE_L2CC_MUTEX_H
+#ifndef __KERNEL_TEE_L2CC_MUTEX_H
+#define __KERNEL_TEE_L2CC_MUTEX_H
 
-#endif /* TEE_L2CC_MUTEX_H */
+#endif /* __KERNEL_TEE_L2CC_MUTEX_H */
diff --git a/core/arch/riscv/include/kernel/tlb_helpers.h b/core/arch/riscv/include/kernel/tlb_helpers.h
index ad82024ab..d81766b9d 100644
--- a/core/arch/riscv/include/kernel/tlb_helpers.h
+++ b/core/arch/riscv/include/kernel/tlb_helpers.h
@@ -3,8 +3,8 @@
  * Copyright 2022 NXP
  */
 
-#ifndef TLB_HELPERS_H
-#define TLB_HELPERS_H
+#ifndef __KERNEL_TLB_HELPERS_H
+#define __KERNEL_TLB_HELPERS_H
 
 #ifndef __ASSEMBLER__
 
@@ -15,4 +15,4 @@ void tlbi_va_asid(vaddr_t va, uint32_t asid);
 
 #endif /*!__ASSEMBLER__*/
 
-#endif /* TLB_HELPERS_H */
+#endif /* __KERNEL_TLB_HELPERS_H */
diff --git a/core/arch/riscv/include/mm/core_mmu_arch.h b/core/arch/riscv/include/mm/core_mmu_arch.h
index 1284d4742..cc9aa798f 100644
--- a/core/arch/riscv/include/mm/core_mmu_arch.h
+++ b/core/arch/riscv/include/mm/core_mmu_arch.h
@@ -2,8 +2,8 @@
 /*
  * Copyright 2022-2023 NXP
  */
-#ifndef __CORE_MMU_ARCH_H
-#define __CORE_MMU_ARCH_H
+#ifndef __MM_CORE_MMU_ARCH_H
+#define __MM_CORE_MMU_ARCH_H
 
 #ifndef __ASSEMBLER__
 #include <assert.h>
@@ -150,4 +150,4 @@ static inline bool core_mmu_level_in_range(unsigned int level)
 }
 #endif /*__ASSEMBLER__*/
 
-#endif /* __CORE_MMU_ARCH_H */
+#endif /* __MM_CORE_MMU_ARCH_H */
diff --git a/core/arch/riscv/include/tee/optee_abi.h b/core/arch/riscv/include/tee/optee_abi.h
index e65bebcac..8528b34ca 100644
--- a/core/arch/riscv/include/tee/optee_abi.h
+++ b/core/arch/riscv/include/tee/optee_abi.h
@@ -3,8 +3,8 @@
  * Copyright 2023 NXP
  * Copyright (c) 2015-2021, Linaro Limited
  */
-#ifndef OPTEE_ABI_H
-#define OPTEE_ABI_H
+#ifndef __TEE_OPTEE_ABI_H
+#define __TEE_OPTEE_ABI_H
 
 #include <stdint.h>
 
@@ -702,4 +702,4 @@
 	((((ret) & OPTEE_ABI_RETURN_RPC_PREFIX_MASK) == \
 		OPTEE_ABI_RETURN_RPC_PREFIX))); })
 
-#endif /* OPTEE_ABI_H */
+#endif /* __TEE_OPTEE_ABI_H */
diff --git a/core/arch/riscv/include/tee/teeabi_opteed.h b/core/arch/riscv/include/tee/teeabi_opteed.h
index 50f846548..05c6aabb8 100644
--- a/core/arch/riscv/include/tee/teeabi_opteed.h
+++ b/core/arch/riscv/include/tee/teeabi_opteed.h
@@ -4,8 +4,8 @@
  * Copyright (c) 2014, Linaro Limited
  */
 
-#ifndef TEEABI_OPTEED_H
-#define TEEABI_OPTEED_H
+#ifndef __TEE_TEEABI_OPTEED_H
+#define __TEE_TEEABI_OPTEED_H
 
 /*
  * This file specify ABI function IDs used when returning from TEE to the
@@ -108,4 +108,4 @@
 #define TEEABI_OPTEED_RETURN_SYSTEM_RESET_DONE \
 	TEEABI_OPTEED_RV(TEEABI_OPTEED_FUNCID_RETURN_SYSTEM_RESET_DONE)
 
-#endif /*TEEABI_OPTEED_H*/
+#endif /*__TEE_TEEABI_OPTEED_H*/
diff --git a/core/arch/riscv/include/tee/teeabi_opteed_macros.h b/core/arch/riscv/include/tee/teeabi_opteed_macros.h
index 6345bd613..911821d5b 100644
--- a/core/arch/riscv/include/tee/teeabi_opteed_macros.h
+++ b/core/arch/riscv/include/tee/teeabi_opteed_macros.h
@@ -4,11 +4,11 @@
  * Copyright (c) 2014, Linaro Limited
  */
 
-#ifndef TEEABI_OPTEED_MACROS_H
-#define TEEABI_OPTEED_MACROS_H
+#ifndef __TEE_TEEABI_OPTEED_MACROS_H
+#define __TEE_TEEABI_OPTEED_MACROS_H
 
 #define TEEABI_OPTEED_RV(func_num) \
 	OPTEE_ABI_CALL_VAL(OPTEE_ABI_32, OPTEE_ABI_FAST_CALL, \
 			   OPTEE_ABI_OWNER_TRUSTED_OS_OPTEED, (func_num))
 
-#endif /*TEEABI_OPTEED_MACROS_H*/
+#endif /*__TEE_TEEABI_OPTEED_MACROS_H*/
diff --git a/core/include/crypto/crypto_se.h b/core/include/crypto/crypto_se.h
index eb7452faa..d8471e51a 100644
--- a/core/include/crypto/crypto_se.h
+++ b/core/include/crypto/crypto_se.h
@@ -10,8 +10,8 @@
   * These requests shall be handled in the secure element normally placed on
   * a serial communication bus (SPI, I2C).
   */
-#ifndef __CRYPTO_SE_H
-#define __CRYPTO_SE_H
+#ifndef __CRYPTO_CRYPTO_SE_H
+#define __CRYPTO_CRYPTO_SE_H
 
 #include <tee_api_types.h>
 
diff --git a/core/include/drivers/amlogic_uart.h b/core/include/drivers/amlogic_uart.h
index 97b55f226..e575afeab 100644
--- a/core/include/drivers/amlogic_uart.h
+++ b/core/include/drivers/amlogic_uart.h
@@ -1,7 +1,7 @@
 /* SPDX-License-Identifier: BSD-2-Clause */
 
-#ifndef AMLOGIC_UART_H
-#define AMLOGIC_UART_H
+#ifndef __DRIVERS_AMLOGIC_UART_H
+#define __DRIVERS_AMLOGIC_UART_H
 
 #include <types_ext.h>
 #include <drivers/serial.h>
@@ -13,4 +13,4 @@ struct amlogic_uart_data {
 
 void amlogic_uart_init(struct amlogic_uart_data *pd, paddr_t base);
 
-#endif /* AMLOGIC_UART_H */
+#endif /* __DRIVERS_AMLOGIC_UART_H */
diff --git a/core/include/drivers/atmel_uart.h b/core/include/drivers/atmel_uart.h
index aa5563f70..2639fce72 100644
--- a/core/include/drivers/atmel_uart.h
+++ b/core/include/drivers/atmel_uart.h
@@ -25,8 +25,8 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
-#ifndef ATMEL_UART_H
-#define ATMEL_UART_H
+#ifndef __DRIVERS_ATMEL_UART_H
+#define __DRIVERS_ATMEL_UART_H
 
 #include <types_ext.h>
 #include <drivers/serial.h>
@@ -38,4 +38,4 @@ struct atmel_uart_data {
 
 void atmel_uart_init(struct atmel_uart_data *pd, paddr_t base);
 
-#endif /* ATMEL_UART_H */
+#endif /* __DRIVERS_ATMEL_UART_H */
diff --git a/core/include/drivers/bcm/bnxt.h b/core/include/drivers/bcm/bnxt.h
index 7136c7efa..51aae18d9 100644
--- a/core/include/drivers/bcm/bnxt.h
+++ b/core/include/drivers/bcm/bnxt.h
@@ -3,8 +3,8 @@
  * Copyright 2019 Broadcom.
  */
 
-#ifndef BNXT_H
-#define BNXT_H
+#ifndef __DRIVERS_BCM_BNXT_H
+#define __DRIVERS_BCM_BNXT_H
 
 #include <compiler.h>
 #include <io.h>
diff --git a/core/include/drivers/bcm_gpio.h b/core/include/drivers/bcm_gpio.h
index 6606656bf..7d5e408af 100644
--- a/core/include/drivers/bcm_gpio.h
+++ b/core/include/drivers/bcm_gpio.h
@@ -3,8 +3,8 @@
  * Copyright 2019 Broadcom.
  */
 
-#ifndef BCM_GPIO_H
-#define BCM_GPIO_H
+#ifndef __DRIVERS_BCM_GPIO_H
+#define __DRIVERS_BCM_GPIO_H
 
 #include <drivers/gpio.h>
 #include <stdlib.h>
@@ -30,4 +30,4 @@ struct bcm_gpio_chip {
 struct bcm_gpio_chip *bcm_gpio_pin_to_chip(unsigned int pin);
 /* Set gpiopin as secure */
 void iproc_gpio_set_secure(int gpiopin);
-#endif	/* BCM_GPIO_H */
+#endif	/* __DRIVERS_BCM_GPIO_H */
diff --git a/core/include/drivers/bcm_hwrng.h b/core/include/drivers/bcm_hwrng.h
index abf213df6..b4d84f967 100644
--- a/core/include/drivers/bcm_hwrng.h
+++ b/core/include/drivers/bcm_hwrng.h
@@ -3,11 +3,11 @@
  * Copyright 2019 Broadcom.
  */
 
-#ifndef BCM_HWRNG_H
-#define BCM_HWRNG_H
+#ifndef __DRIVERS_BCM_HWRNG_H
+#define __DRIVERS_BCM_HWRNG_H
 
 #include <stdlib.h>
 
 uint32_t bcm_hwrng_read_rng(uint32_t *p_out, uint32_t words_to_read);
 
-#endif /* BCM_HWRNG_H */
+#endif /* __DRIVERS_BCM_HWRNG_H */
diff --git a/core/include/drivers/bcm_sotp.h b/core/include/drivers/bcm_sotp.h
index 247600425..aa1327043 100644
--- a/core/include/drivers/bcm_sotp.h
+++ b/core/include/drivers/bcm_sotp.h
@@ -3,8 +3,8 @@
  * Copyright 2019 Broadcom.
  */
 
-#ifndef BCM_SOTP_H
-#define BCM_SOTP_H
+#ifndef __DRIVERS_BCM_SOTP_H
+#define __DRIVERS_BCM_SOTP_H
 
 #include <stdint.h>
 #include <tee_api.h>
diff --git a/core/include/drivers/caam_extension.h b/core/include/drivers/caam_extension.h
index d0232176e..41ccec056 100644
--- a/core/include/drivers/caam_extension.h
+++ b/core/include/drivers/caam_extension.h
@@ -52,4 +52,4 @@ TEE_Result caam_mp_sign(uint8_t *data, size_t *data_size, uint8_t *sig,
 TEE_Result caam_dek_generate(const uint8_t *payload, size_t payload_size,
 			     uint8_t *dek, size_t dek_size);
 #endif /* CFG_NXP_CAAM_DEK_DRV */
-#endif /* __CAAM_EXTENSION_H__ */
+#endif /* __DRIVERS_CAAM_EXTENSION_H__ */
diff --git a/core/include/drivers/cdns_uart.h b/core/include/drivers/cdns_uart.h
index 19516ff86..b5ee11234 100644
--- a/core/include/drivers/cdns_uart.h
+++ b/core/include/drivers/cdns_uart.h
@@ -26,8 +26,8 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
-#ifndef CDNS_UART_H
-#define CDNS_UART_H
+#ifndef __DRIVERS_CDNS_UART_H
+#define __DRIVERS_CDNS_UART_H
 
 #include <types_ext.h>
 #include <drivers/serial.h>
@@ -40,4 +40,4 @@ struct cdns_uart_data {
 void cdns_uart_init(struct cdns_uart_data *pd, paddr_t base, uint32_t uart_clk,
 		uint32_t baud_rate);
 
-#endif /* CDNS_UART_H */
+#endif /* __DRIVERS_CDNS_UART_H */
diff --git a/core/include/drivers/hi16xx_uart.h b/core/include/drivers/hi16xx_uart.h
index 83413fc0e..9fb11a4f5 100644
--- a/core/include/drivers/hi16xx_uart.h
+++ b/core/include/drivers/hi16xx_uart.h
@@ -7,8 +7,8 @@
  * UART driver for Hisilicon Hi16xx and Phosphor V660 (hip05) SoCs
  */
 
-#ifndef HI16XX_UART_H
-#define HI16XX_UART_H
+#ifndef __DRIVERS_HI16XX_UART_H
+#define __DRIVERS_HI16XX_UART_H
 
 #include <types_ext.h>
 #include <drivers/serial.h>
@@ -23,4 +23,4 @@ struct hi16xx_uart_data {
 void hi16xx_uart_init(struct hi16xx_uart_data *pd, paddr_t base,
 		      uint32_t uart_clk, uint32_t baud_rate);
 
-#endif /* HI16XX_UART_H */
+#endif /* __DRIVERS_HI16XX_UART_H */
diff --git a/core/include/drivers/imx/dcp.h b/core/include/drivers/imx/dcp.h
index a01ae3011..28ff08031 100644
--- a/core/include/drivers/imx/dcp.h
+++ b/core/include/drivers/imx/dcp.h
@@ -2,8 +2,8 @@
 /*
  * Copyright 2020 NXP
  */
-#ifndef __IMX_DCP_H__
-#define __IMX_DCP_H__
+#ifndef __DRIVERS_IMX_DCP_H
+#define __DRIVERS_IMX_DCP_H
 
 #include <compiler.h>
 #include <tee_api_types.h>
@@ -192,4 +192,4 @@ static inline TEE_Result dcp_vbase(vaddr_t *base __unused)
 }
 #endif /* CFG_DT */
 
-#endif /* __IMX_DCP_H__ */
+#endif /* __DRIVERS_IMX_DCP_H */
diff --git a/core/include/drivers/imx_ocotp.h b/core/include/drivers/imx_ocotp.h
index 72d4b7960..d10013515 100644
--- a/core/include/drivers/imx_ocotp.h
+++ b/core/include/drivers/imx_ocotp.h
@@ -3,8 +3,8 @@
  * Copyright 2021 NXP
  */
 
-#ifndef _IMX_OCOTP_H
-#define _IMX_OCOTP_H
+#ifndef __DRIVERS_IMX_OCOTP_H
+#define __DRIVERS_IMX_OCOTP_H
 
 #include <tee_api_types.h>
 
@@ -19,4 +19,4 @@
  * @[out]val Shadow register value
  */
 TEE_Result imx_ocotp_read(unsigned int bank, unsigned int word, uint32_t *val);
-#endif /* _IMX_OCOTP_H */
+#endif /* __DRIVERS_IMX_OCOTP_H */
diff --git a/core/include/drivers/imx_uart.h b/core/include/drivers/imx_uart.h
index b9df517e5..a62bd17a2 100644
--- a/core/include/drivers/imx_uart.h
+++ b/core/include/drivers/imx_uart.h
@@ -26,8 +26,8 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
-#ifndef IMX_UART_H
-#define IMX_UART_H
+#ifndef __DRIVERS_IMX_UART_H
+#define __DRIVERS_IMX_UART_H
 
 #include <types_ext.h>
 #include <drivers/serial.h>
@@ -39,4 +39,4 @@ struct imx_uart_data {
 
 void imx_uart_init(struct imx_uart_data *pd, paddr_t base);
 
-#endif /* IMX_UART_H */
+#endif /* __DRIVERS_IMX_UART_H */
diff --git a/core/include/drivers/imx_wdog.h b/core/include/drivers/imx_wdog.h
index 2c7607335..7df879aab 100644
--- a/core/include/drivers/imx_wdog.h
+++ b/core/include/drivers/imx_wdog.h
@@ -25,8 +25,8 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#ifndef __IMX_WDOG_H
-#define __IMX_WDOG_H
+#ifndef __DRIVERS_IMX_WDOG_H
+#define __DRIVERS_IMX_WDOG_H
 
 #include <compiler.h>
 
diff --git a/core/include/drivers/lpc_uart.h b/core/include/drivers/lpc_uart.h
index 3e235b4e2..e219a90b2 100644
--- a/core/include/drivers/lpc_uart.h
+++ b/core/include/drivers/lpc_uart.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2022, HiSilicon Limited
  */
 
-#ifndef LPC_UART_H
-#define LPC_UART_H
+#ifndef __DRIVERS_LPC_UART_H
+#define __DRIVERS_LPC_UART_H
 
 #include <types_ext.h>
 #include <drivers/serial.h>
@@ -50,4 +50,4 @@ struct lpc_uart_data {
 void lpc_uart_init(struct lpc_uart_data *pd, paddr_t base,
 		   uint32_t uart_clk, uint32_t baud_rate);
 
-#endif /* LPC_UART_H */
+#endif /* __DRIVERS_LPC_UART_H */
diff --git a/core/include/drivers/mvebu_uart.h b/core/include/drivers/mvebu_uart.h
index fed9ac8b6..a39342875 100644
--- a/core/include/drivers/mvebu_uart.h
+++ b/core/include/drivers/mvebu_uart.h
@@ -25,8 +25,8 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
-#ifndef MVEBU_UART_H
-#define MVEBU_UART_H
+#ifndef __DRIVERS_MVEBU_UART_H
+#define __DRIVERS_MVEBU_UART_H
 
 #include <types_ext.h>
 #include <drivers/serial.h>
@@ -39,4 +39,4 @@ struct mvebu_uart_data {
 void mvebu_uart_init(struct mvebu_uart_data *pd, paddr_t pbase,
 		uint32_t uart_clk, uint32_t baud_rate);
 
-#endif /* MVEBU_UART_H */
+#endif /* __DRIVERS_MVEBU_UART_H */
diff --git a/core/include/drivers/ns16550.h b/core/include/drivers/ns16550.h
index de0be9eb3..60d367129 100644
--- a/core/include/drivers/ns16550.h
+++ b/core/include/drivers/ns16550.h
@@ -26,8 +26,8 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
-#ifndef NS16550_H
-#define NS16550_H
+#ifndef __DRIVERS_NS16550_H
+#define __DRIVERS_NS16550_H
 
 #include <drivers/serial.h>
 #include <io.h>
@@ -64,4 +64,4 @@ static inline void serial_out(vaddr_t addr, uint8_t io_width, int ch)
 void ns16550_init(struct ns16550_data *pd, paddr_t base, uint8_t io_width,
 		  uint8_t reg_shift);
 
-#endif /* NS16550_H */
+#endif /* __DRIVERS_NS16550_H */
diff --git a/core/include/drivers/pl011.h b/core/include/drivers/pl011.h
index b46880eeb..59475c1c2 100644
--- a/core/include/drivers/pl011.h
+++ b/core/include/drivers/pl011.h
@@ -2,8 +2,8 @@
 /*
  * Copyright (c) 2014, Linaro Limited
  */
-#ifndef PL011_H
-#define PL011_H
+#ifndef __DRIVERS_PL011_H
+#define __DRIVERS_PL011_H
 
 #include <types_ext.h>
 #include <drivers/serial.h>
@@ -18,4 +18,4 @@ struct pl011_data {
 void pl011_init(struct pl011_data *pd, paddr_t pbase, uint32_t uart_clk,
 		uint32_t baud_rate);
 
-#endif /* PL011_H */
+#endif /* __DRIVERS_PL011_H */
diff --git a/core/include/drivers/pl022_spi.h b/core/include/drivers/pl022_spi.h
index c20b01be2..0ccef78a2 100644
--- a/core/include/drivers/pl022_spi.h
+++ b/core/include/drivers/pl022_spi.h
@@ -4,8 +4,8 @@
  *
  */
 
-#ifndef __PL022_SPI_H__
-#define __PL022_SPI_H__
+#ifndef __DRIVERS_PL022_SPI_H
+#define __DRIVERS_PL022_SPI_H
 
 #include <drivers/gpio.h>
 #include <spi.h>
@@ -42,5 +42,5 @@ struct pl022_data {
 
 void pl022_init(struct pl022_data *pd);
 
-#endif	/* __PL022_SPI_H__ */
+#endif	/* __DRIVERS_PL022_SPI_H */
 
diff --git a/core/include/drivers/pl061_gpio.h b/core/include/drivers/pl061_gpio.h
index 1676b2f22..d6a36c41e 100644
--- a/core/include/drivers/pl061_gpio.h
+++ b/core/include/drivers/pl061_gpio.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2016, Linaro Limited
  */
 
-#ifndef __PL061_GPIO_H__
-#define __PL061_GPIO_H__
+#ifndef __DRIVERS_PL061_GPIO_H
+#define __DRIVERS_PL061_GPIO_H
 
 #include <drivers/gpio.h>
 #include <types_ext.h>
@@ -26,4 +26,4 @@ enum pl061_mode_control pl061_get_mode_control(unsigned int gpio_pin);
 void pl061_set_mode_control(unsigned int gpio_pin,
 	enum pl061_mode_control hw_sw);
 
-#endif	/* __PL061_GPIO_H__ */
+#endif	/* __DRIVERS_PL061_GPIO_H */
diff --git a/core/include/drivers/sam/at91_ddr.h b/core/include/drivers/sam/at91_ddr.h
index 84e13e630..5424303ba 100644
--- a/core/include/drivers/sam/at91_ddr.h
+++ b/core/include/drivers/sam/at91_ddr.h
@@ -6,8 +6,8 @@
  *	Nicolas Ferre <nicolas.ferre@atmel.com>
  */
 
-#ifndef AT91_DDRSDR_H
-#define AT91_DDRSDR_H
+#ifndef __DRIVERS_SAM_AT91_DDRSDR_H
+#define __DRIVERS_SAM_AT91_DDRSDR_H
 
 /* Mode Register */
 #define AT91_DDRSDRC_MR				0x00
diff --git a/core/include/drivers/scif.h b/core/include/drivers/scif.h
index fe8187159..7cb3ac015 100644
--- a/core/include/drivers/scif.h
+++ b/core/include/drivers/scif.h
@@ -26,8 +26,8 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
-#ifndef SCIF_H
-#define SCIF_H
+#ifndef __DRIVERS_SCIF_H
+#define __DRIVERS_SCIF_H
 
 #include <types_ext.h>
 #include <drivers/serial.h>
@@ -41,4 +41,4 @@ struct scif_uart_data {
 
 void scif_uart_init(struct scif_uart_data *pd, paddr_t base);
 
-#endif /* SCIF */
+#endif /* __DRIVERS_SCIF_H */
diff --git a/core/include/drivers/scmi-msg.h b/core/include/drivers/scmi-msg.h
index 08dd31c67..28961712f 100644
--- a/core/include/drivers/scmi-msg.h
+++ b/core/include/drivers/scmi-msg.h
@@ -4,8 +4,8 @@
  * Copyright (c) 2019-2021, Linaro Limited
  */
 
-#ifndef SCMI_MSG_H
-#define SCMI_MSG_H
+#ifndef __DRIVERS_SCMI_MSG_H
+#define __DRIVERS_SCMI_MSG_H
 
 #include <compiler.h>
 #include <kernel/panic.h>
@@ -408,4 +408,4 @@ int32_t plat_scmi_voltd_get_config(unsigned int channel_id,
 int32_t plat_scmi_voltd_set_config(unsigned int channel_id,
 				   unsigned int scmi_id, uint32_t config);
 
-#endif /* SCMI_MSG_H */
+#endif /* __DRIVERS_SCMI_MSG_H */
diff --git a/core/include/drivers/scmi.h b/core/include/drivers/scmi.h
index 9497206b3..1e43bac6c 100644
--- a/core/include/drivers/scmi.h
+++ b/core/include/drivers/scmi.h
@@ -2,8 +2,8 @@
 /*
  * Copyright (c) 2015-2019, Arm Limited and Contributors. All rights reserved.
  */
-#ifndef SCMI_MSG_SCMI_H
-#define SCMI_MSG_SCMI_H
+#ifndef __DRIVERS_SCMI_MSG_SCMI_H
+#define __DRIVERS_SCMI_MSG_SCMI_H
 
 #define SCMI_PROTOCOL_ID_BASE			0x10
 #define SCMI_PROTOCOL_ID_POWER_DOMAIN		0x11
@@ -27,4 +27,4 @@
 #define SCMI_HARDWARE_ERROR		(-9)
 #define SCMI_PROTOCOL_ERROR		(-10)
 
-#endif /* SCMI_MSG_SCMI_H */
+#endif /* __DRIVERS_SCMI_MSG_SCMI_H */
diff --git a/core/include/drivers/serial8250_uart.h b/core/include/drivers/serial8250_uart.h
index 40816c9b9..e56811fb4 100644
--- a/core/include/drivers/serial8250_uart.h
+++ b/core/include/drivers/serial8250_uart.h
@@ -2,8 +2,8 @@
 /*
  * Copyright (c) 2015, Linaro Limited
  */
-#ifndef SERIAL8250_UART_H
-#define SERIAL8250_UART_H
+#ifndef __DRIVERS_SERIAL8250_UART_H
+#define __DRIVERS_SERIAL8250_UART_H
 
 #include <types_ext.h>
 #include <drivers/serial.h>
@@ -18,5 +18,5 @@ struct serial8250_uart_data {
 void serial8250_uart_init(struct serial8250_uart_data *pd, paddr_t base,
 			  uint32_t uart_clk, uint32_t baud_rate);
 
-#endif /* SERIAL8250_UART_H */
+#endif /* __DRIVERS_SERIAL8250_UART_H */
 
diff --git a/core/include/drivers/sp805_wdt.h b/core/include/drivers/sp805_wdt.h
index 0d5d3133b..a8f8f4573 100644
--- a/core/include/drivers/sp805_wdt.h
+++ b/core/include/drivers/sp805_wdt.h
@@ -3,8 +3,8 @@
  * Copyright 2019 Broadcom.
  */
 
-#ifndef SP805_WDT_H
-#define SP805_WDT_H
+#ifndef __DRIVERS_SP805_WDT_H
+#define __DRIVERS_SP805_WDT_H
 
 #include <drivers/wdt.h>
 #include <kernel/interrupt.h>
@@ -66,4 +66,4 @@ TEE_Result sp805_register_itr_handler(struct sp805_wdt_data *pd,
 				      uint32_t itr_num, uint32_t itr_flag,
 				      sp805_itr_handler_func_t itr_handler);
 
-#endif /* SP805_WDT_H */
+#endif /* __DRIVERS_SP805_WDT_H */
diff --git a/core/include/drivers/sprd_uart.h b/core/include/drivers/sprd_uart.h
index 64826222d..8c12c3955 100644
--- a/core/include/drivers/sprd_uart.h
+++ b/core/include/drivers/sprd_uart.h
@@ -26,8 +26,8 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
-#ifndef SPRD_UART_H
-#define SPRD_UART_H
+#ifndef __DRIVERS_SPRD_UART_H
+#define __DRIVERS_SPRD_UART_H
 
 #include <types_ext.h>
 #include <drivers/serial.h>
@@ -39,5 +39,5 @@ struct sprd_uart_data {
 
 void sprd_uart_init(struct sprd_uart_data *pd, paddr_t base);
 
-#endif /* SPRD_UART_H */
+#endif /* __DRIVERS_SPRD_UART_H */
 
diff --git a/core/include/drivers/stih_asc.h b/core/include/drivers/stih_asc.h
index 21e829766..79df157b8 100644
--- a/core/include/drivers/stih_asc.h
+++ b/core/include/drivers/stih_asc.h
@@ -2,8 +2,8 @@
 /*
  * Copyright (c) 2017, Linaro Limited
  */
-#ifndef STIH_ASC_H
-#define STIH_ASC_H
+#ifndef __DRIVERS_STIH_ASC_H
+#define __DRIVERS_STIH_ASC_H
 
 #include <drivers/serial.h>
 #include <types_ext.h>
@@ -17,5 +17,5 @@ struct stih_asc_pd {
 
 void stih_asc_init(struct stih_asc_pd *pb, vaddr_t base);
 
-#endif /* STIH_ASC_H */
+#endif /* __DRIVERS_STIH_ASC_H */
 
diff --git a/core/include/drivers/stm32_bsec.h b/core/include/drivers/stm32_bsec.h
index a59842582..f0577bc8b 100644
--- a/core/include/drivers/stm32_bsec.h
+++ b/core/include/drivers/stm32_bsec.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2017-2022, STMicroelectronics
  */
 
-#ifndef __STM32_BSEC_H
-#define __STM32_BSEC_H
+#ifndef __DRIVERS_STM32_BSEC_H
+#define __DRIVERS_STM32_BSEC_H
 
 #include <compiler.h>
 #include <stdint.h>
@@ -199,4 +199,4 @@ TEE_Result stm32_bsec_find_otp_by_phandle(const uint32_t phandle,
  */
 TEE_Result stm32_bsec_get_state(enum stm32_bsec_sec_state *sec_state);
 
-#endif /*__STM32_BSEC_H*/
+#endif /*__DRIVERS_STM32_BSEC_H*/
diff --git a/core/include/drivers/stm32_etzpc.h b/core/include/drivers/stm32_etzpc.h
index 3a87ff45a..48f0385a3 100644
--- a/core/include/drivers/stm32_etzpc.h
+++ b/core/include/drivers/stm32_etzpc.h
@@ -4,8 +4,8 @@
  * Copyright (c) 2018-2019, STMicroelectronics
  */
 
-#ifndef __STM32_ETZPC_H__
-#define __STM32_ETZPC_H__
+#ifndef __DRIVERS_STM32_ETZPC_H
+#define __DRIVERS_STM32_ETZPC_H
 
 #include <util.h>
 #include <types_ext.h>
@@ -74,4 +74,4 @@ void etzpc_lock_tzma(uint32_t tzma_id);
  * Return true if TZMA is locked, false otherwise
  */
 bool etzpc_get_lock_tzma(uint32_t tzma_id);
-#endif /*__STM32_ETZPC_H__*/
+#endif /*__DRIVERS_STM32_ETZPC_H*/
diff --git a/core/include/drivers/stm32mp_dt_bindings.h b/core/include/drivers/stm32mp_dt_bindings.h
index 2d7a27cf2..1c438ecef 100644
--- a/core/include/drivers/stm32mp_dt_bindings.h
+++ b/core/include/drivers/stm32mp_dt_bindings.h
@@ -2,8 +2,8 @@
 /*
  * Copyright (c) 2022, STMicroelectronics
  */
-#ifndef STM32MP_DT_BINDINGS_H
-#define STM32MP_DT_BINDINGS_H
+#ifndef __DRIVERS_STM32MP_DT_BINDINGS_H
+#define __DRIVERS_STM32MP_DT_BINDINGS_H
 
 #ifdef CFG_STM32MP13
 #include <dt-bindings/clock/stm32mp13-clks.h>
@@ -18,4 +18,4 @@
 #include <dt-bindings/reset/stm32mp1-resets.h>
 #endif
 
-#endif /* STM32MP_DT_BINDINGS_H */
+#endif /* __DRIVERS_STM32MP_DT_BINDINGS_H */
diff --git a/core/include/drivers/stpmic1.h b/core/include/drivers/stpmic1.h
index 0d020caed..13539c658 100644
--- a/core/include/drivers/stpmic1.h
+++ b/core/include/drivers/stpmic1.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2016-2018, STMicroelectronics - All Rights Reserved
  */
 
-#ifndef __STPMIC1_H__
-#define __STPMIC1_H__
+#ifndef __DRIVERS_STPMIC1_H
+#define __DRIVERS_STPMIC1_H
 
 #include <drivers/stm32_i2c.h>
 #include <util.h>
@@ -241,4 +241,4 @@ int stpmic1_lp_voltage_cfg(const char *name, uint16_t millivolts,
 			   struct stpmic1_lp_cfg *cfg);
 int stpmic1_lp_voltage_unpg(struct stpmic1_lp_cfg *cfg);
 
-#endif /*__STPMIC1_H__*/
+#endif /*__DRIVERS_STPMIC1_H*/
diff --git a/core/include/kernel/chip_services.h b/core/include/kernel/chip_services.h
index a22237ce1..c051078f1 100644
--- a/core/include/kernel/chip_services.h
+++ b/core/include/kernel/chip_services.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2014, STMicroelectronics International N.V.
  */
 
-#ifndef CHIP_SERVICES_H
-#define CHIP_SERVICES_H
+#ifndef __KERNEL_CHIP_SERVICES_H
+#define __KERNEL_CHIP_SERVICES_H
 
 /* Die ID */
 #define FVR_DIE_ID_NUM_REGS                 3
diff --git a/core/include/kernel/dt_driver.h b/core/include/kernel/dt_driver.h
index 38a934ad1..989be5a1a 100644
--- a/core/include/kernel/dt_driver.h
+++ b/core/include/kernel/dt_driver.h
@@ -5,8 +5,8 @@
  * Copyright (c) 2021, STMicroelectronics
  */
 
-#ifndef __DT_DRIVER_H
-#define __DT_DRIVER_H
+#ifndef __KERNEL_DT_DRIVER_H
+#define __KERNEL_DT_DRIVER_H
 
 #include <kernel/dt.h>
 #include <stdint.h>
@@ -278,5 +278,4 @@ static inline TEE_Result dt_driver_test_status(void)
 	return TEE_ERROR_NOT_IMPLEMENTED;
 }
 #endif
-
-#endif /* __DT_DRIVER_H */
+#endif /* __KERNEL_DT_DRIVER_H */
diff --git a/core/include/kernel/rpc_io_i2c.h b/core/include/kernel/rpc_io_i2c.h
index 50ddbcd4d..fe9f6b8e6 100644
--- a/core/include/kernel/rpc_io_i2c.h
+++ b/core/include/kernel/rpc_io_i2c.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2020 Foundries Ltd <jorge@foundries.io>
  */
 
-#ifndef __RPC_IO_I2C_H
-#define __RPC_IO_I2C_H
+#ifndef __KERNEL_RPC_IO_I2C_H
+#define __KERNEL_RPC_IO_I2C_H
 
 #include <optee_rpc_cmd.h>
 #include <tee_api_types.h>
@@ -35,4 +35,4 @@ struct rpc_i2c_request {
 
 TEE_Result rpc_io_i2c_transfer(struct rpc_i2c_request *p, size_t *bytes);
 
-#endif /* __RPC_IO_I2C_H */
+#endif /* __KERNEL_RPC_IO_I2C_H */
diff --git a/core/include/kernel/tee_common.h b/core/include/kernel/tee_common.h
index 5c5da2d6e..d4b067210 100644
--- a/core/include/kernel/tee_common.h
+++ b/core/include/kernel/tee_common.h
@@ -2,8 +2,8 @@
 /*
  * Copyright (c) 2014, STMicroelectronics International N.V.
  */
-#ifndef TEE_COMMON_H
-#define TEE_COMMON_H
+#ifndef __KERNEL_TEE_COMMON_H
+#define __KERNEL_TEE_COMMON_H
 
 #include <stdlib.h>
 
@@ -35,4 +35,4 @@ void tee_mtime_perftest(void);
 #define tee_mtime_perftest()  do { } while (0)
 #endif
 
-#endif /* TEE_COMMON_H */
+#endif /* __KERNEL_TEE_COMMON_H */
diff --git a/core/include/kernel/tee_common_otp.h b/core/include/kernel/tee_common_otp.h
index 2b0d3a3a2..cc1f0a88b 100644
--- a/core/include/kernel/tee_common_otp.h
+++ b/core/include/kernel/tee_common_otp.h
@@ -2,8 +2,8 @@
 /*
  * Copyright (c) 2014, STMicroelectronics International N.V.
  */
-#ifndef TEE_COMMON_OTP_H
-#define TEE_COMMON_OTP_H
+#ifndef __KERNEL_TEE_COMMON_OTP_H
+#define __KERNEL_TEE_COMMON_OTP_H
 
 #include <stddef.h>
 #include <stdint.h>
@@ -20,4 +20,4 @@ int tee_otp_get_die_id(uint8_t *buffer, size_t len);
 TEE_Result tee_otp_get_ta_enc_key(uint32_t key_type, uint8_t *buffer,
 				  size_t len);
 
-#endif /* TEE_COMMON_OTP_H */
+#endif /* __KERNEL_TEE_COMMON_OTP_H */
diff --git a/core/include/kernel/tee_misc.h b/core/include/kernel/tee_misc.h
index 035acd37f..665525202 100644
--- a/core/include/kernel/tee_misc.h
+++ b/core/include/kernel/tee_misc.h
@@ -2,8 +2,8 @@
 /*
  * Copyright (c) 2014, STMicroelectronics International N.V.
  */
-#ifndef TEE_MISC_H
-#define TEE_MISC_H
+#ifndef __KERNEL_TEE_MISC_H
+#define __KERNEL_TEE_MISC_H
 
 #include <types_ext.h>
 
@@ -68,4 +68,4 @@ bool core_is_buffer_intersect(paddr_t b, paddr_size_t bl,
  */
 void *alloc_cache_aligned(size_t size);
 
-#endif /* TEE_MISC_H */
+#endif /* __KERNEL_TEE_MISC_H */
diff --git a/core/include/kernel/tee_ta_manager.h b/core/include/kernel/tee_ta_manager.h
index ef689b520..8cae35002 100644
--- a/core/include/kernel/tee_ta_manager.h
+++ b/core/include/kernel/tee_ta_manager.h
@@ -5,8 +5,8 @@
  * Copyright (c) 2020, Arm Limited
  */
 
-#ifndef TEE_TA_MANAGER_H
-#define TEE_TA_MANAGER_H
+#ifndef __KERNEL_TEE_TA_MANAGER_H
+#define __KERNEL_TEE_TA_MANAGER_H
 
 #include <assert.h>
 #include <kernel/mutex.h>
diff --git a/core/include/kernel/tee_time.h b/core/include/kernel/tee_time.h
index 396155104..28b41545c 100644
--- a/core/include/kernel/tee_time.h
+++ b/core/include/kernel/tee_time.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2014, STMicroelectronics International N.V.
  */
 
-#ifndef TEE_TIME_H
-#define TEE_TIME_H
+#ifndef __KERNEL_TEE_TIME_H
+#define __KERNEL_TEE_TIME_H
 
 #include "tee_api_types.h"
 
diff --git a/core/include/kernel/timer.h b/core/include/kernel/timer.h
index 8aaca449a..19abd115b 100644
--- a/core/include/kernel/timer.h
+++ b/core/include/kernel/timer.h
@@ -3,8 +3,8 @@
  * Copyright (C) 2018, Linaro Limited
  */
 
-#ifndef __TIMER_H
-#define __TIMER_H
+#ifndef __KERNEL_TIMER_H
+#define __KERNEL_TIMER_H
 
 void generic_timer_start(uint32_t time_ms);
 void generic_timer_stop(void);
@@ -12,4 +12,4 @@ void generic_timer_stop(void);
 /* Handler for timer expiry interrupt */
 void generic_timer_handler(uint32_t time_ms);
 
-#endif /* __TIMER_H */
+#endif /* __KERNEL_TIMER_H */
diff --git a/core/include/kernel/trace_ta.h b/core/include/kernel/trace_ta.h
index 44644e660..96ced9f37 100644
--- a/core/include/kernel/trace_ta.h
+++ b/core/include/kernel/trace_ta.h
@@ -2,8 +2,8 @@
 /*
  * Copyright (c) 2014, Linaro Limited
  */
-#ifndef TRACE_TA_H
-#define TRACE_TA_H
+#ifndef __KERNEL_TRACE_TA_H
+#define __KERNEL_TRACE_TA_H
 
 #include <trace.h>
 
@@ -16,5 +16,5 @@
 #define TAMSG_RAW(...)
 #endif
 
-#endif /*TRACE_TA_H*/
+#endif /*__KERNEL_TRACE_TA_H*/
 
diff --git a/core/include/mm/core_memprot.h b/core/include/mm/core_memprot.h
index d809ad305..428c827b2 100644
--- a/core/include/mm/core_memprot.h
+++ b/core/include/mm/core_memprot.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2016, Linaro Limited
  * Copyright (c) 2014, STMicroelectronics International N.V.
  */
-#ifndef CORE_MEMPROT_H
-#define CORE_MEMPROT_H
+#ifndef __MM_CORE_MEMPROT_H
+#define __MM_CORE_MEMPROT_H
 
 #include <mm/core_mmu.h>
 #include <types_ext.h>
@@ -115,4 +115,4 @@ vaddr_t io_pa_or_va_secure(struct io_pa_va *p, size_t len);
 vaddr_t io_pa_or_va_nsec(struct io_pa_va *p, size_t len);
 vaddr_t io_pa_or_va(struct io_pa_va *p, size_t len);
 
-#endif /* CORE_MEMPROT_H */
+#endif /* __MM_CORE_MEMPROT_H */
diff --git a/core/include/mm/core_mmu.h b/core/include/mm/core_mmu.h
index c6f258631..be992bf35 100644
--- a/core/include/mm/core_mmu.h
+++ b/core/include/mm/core_mmu.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2016, Linaro Limited
  * Copyright (c) 2014, STMicroelectronics International N.V.
  */
-#ifndef CORE_MMU_H
-#define CORE_MMU_H
+#ifndef __MM_CORE_MMU_H
+#define __MM_CORE_MMU_H
 
 #ifndef __ASSEMBLER__
 #include <assert.h>
@@ -689,4 +689,4 @@ void core_mmu_get_ta_range(paddr_t *base, size_t *size);
 
 #endif /*__ASSEMBLER__*/
 
-#endif /* CORE_MMU_H */
+#endif /* __MM_CORE_MMU_H */
diff --git a/core/include/mm/tee_mm.h b/core/include/mm/tee_mm.h
index 481f58375..b249e4bed 100644
--- a/core/include/mm/tee_mm.h
+++ b/core/include/mm/tee_mm.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2014, STMicroelectronics International N.V.
  */
 
-#ifndef TEE_MM_H
-#define TEE_MM_H
+#ifndef __MM_TEE_MM_H
+#define __MM_TEE_MM_H
 
 #include <malloc.h>
 #include <types_ext.h>
diff --git a/core/include/mm/tee_mmu_types.h b/core/include/mm/tee_mmu_types.h
index e643d27f4..2f495910f 100644
--- a/core/include/mm/tee_mmu_types.h
+++ b/core/include/mm/tee_mmu_types.h
@@ -4,8 +4,8 @@
  * Copyright (c) 2021, Linaro Limited
  * Copyright (c) 2022, Arm Limited.
  */
-#ifndef TEE_MMU_TYPES_H
-#define TEE_MMU_TYPES_H
+#ifndef __MM_TEE_MMU_TYPES_H
+#define __MM_TEE_MMU_TYPES_H
 
 #include <stdint.h>
 #include <sys/queue.h>
diff --git a/core/include/mm/vm.h b/core/include/mm/vm.h
index f1509ba5d..16bafe8f0 100644
--- a/core/include/mm/vm.h
+++ b/core/include/mm/vm.h
@@ -2,8 +2,8 @@
 /*
  * Copyright (c) 2014, STMicroelectronics International N.V.
  */
-#ifndef TEE_MMU_H
-#define TEE_MMU_H
+#ifndef __MM_VM_H
+#define __MM_VM_H
 
 #include <tee_api_types.h>
 #include <kernel/tee_ta_manager.h>
@@ -97,4 +97,4 @@ void vm_set_ctx(struct ts_ctx *ctx);
 
 struct mobj *vm_get_mobj(struct user_mode_ctx *uctx, vaddr_t va, size_t *len,
 			 uint16_t *prot, size_t *offs);
-#endif /*TEE_MMU_H*/
+#endif /*__MM_VM_H*/
diff --git a/core/include/ta_pub_key.h b/core/include/ta_pub_key.h
index 1259e014d..dc02a6f17 100644
--- a/core/include/ta_pub_key.h
+++ b/core/include/ta_pub_key.h
@@ -2,8 +2,8 @@
 /*
  * Copyright (c) 2015, Linaro Limited
  */
-#ifndef KERNEL_TA_PUB_KEY_H
-#define KERNEL_TA_PUB_KEY_H
+#ifndef __TA_PUB_KEY_H
+#define __TA_PUB_KEY_H
 
 #include <types_ext.h>
 
@@ -11,5 +11,5 @@ extern const uint32_t ta_pub_key_exponent;
 extern const uint8_t ta_pub_key_modulus[];
 extern const size_t ta_pub_key_modulus_size;
 
-#endif /*KERNEL_TA_PUB_KEY_H*/
+#endif /*__TA_PUB_KEY_H*/
 
diff --git a/core/include/tee/svc_cache.h b/core/include/tee/svc_cache.h
index a2bccde37..56321edd9 100644
--- a/core/include/tee/svc_cache.h
+++ b/core/include/tee/svc_cache.h
@@ -2,8 +2,8 @@
 /*
  * Copyright (c) 2015, Linaro Limited
  */
-#ifndef SVC_CACHE_H
-#define SVC_CACHE_H
+#ifndef __TEE_SVC_CACHE_H
+#define __TEE_SVC_CACHE_H
 
 #include <types_ext.h>
 #include <tee_api_types.h>
@@ -14,4 +14,4 @@ TEE_Result syscall_cache_operation(void *va, size_t len, unsigned long op);
 #define  syscall_cache_operation syscall_not_supported
 #endif
 
-#endif /*SVC_CACHE_H*/
+#endif /*__TEE_SVC_CACHE_H*/
diff --git a/core/include/tee/tee_cryp_concat_kdf.h b/core/include/tee/tee_cryp_concat_kdf.h
index 49d3348f5..dbcca3135 100644
--- a/core/include/tee/tee_cryp_concat_kdf.h
+++ b/core/include/tee/tee_cryp_concat_kdf.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2014, Linaro Limited
  */
 
-#ifndef TEE_CRYP_CONCAT_KDF_H
-#define TEE_CRYP_CONCAT_KDF_H
+#ifndef __TEE_TEE_CRYP_CONCAT_KDF_H
+#define __TEE_TEE_CRYP_CONCAT_KDF_H
 
 #include <tee_api_types.h>
 
@@ -14,4 +14,4 @@ TEE_Result tee_cryp_concat_kdf(uint32_t hash_id, const uint8_t *shared_secret,
 			       size_t other_info_len, uint8_t *derived_key,
 			       size_t derived_key_len);
 
-#endif /* TEE_CRYP_CONCAT_KDF_H */
+#endif /* __TEE_TEE_CRYP_CONCAT_KDF_H */
diff --git a/core/include/tee/tee_cryp_hkdf.h b/core/include/tee/tee_cryp_hkdf.h
index be0703e42..adce9d893 100644
--- a/core/include/tee/tee_cryp_hkdf.h
+++ b/core/include/tee/tee_cryp_hkdf.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2014, Linaro Limited
  */
 
-#ifndef TEE_CRYP_HKDF_H
-#define TEE_CRYP_HKDF_H
+#ifndef __TEE_TEE_CRYP_HKDF_H
+#define __TEE_TEE_CRYP_HKDF_H
 
 #include <tee_api_types.h>
 
@@ -13,4 +13,4 @@ TEE_Result tee_cryp_hkdf(uint32_t hash_id, const uint8_t *ikm, size_t ikm_len,
 			 const uint8_t *info, size_t info_len, uint8_t *okm,
 			 size_t okm_len);
 
-#endif /* TEE_CRYP_HKDF_H */
+#endif /* __TEE_TEE_CRYP_HKDF_H */
diff --git a/core/include/tee/tee_cryp_pbkdf2.h b/core/include/tee/tee_cryp_pbkdf2.h
index 81b7beb1a..d14103264 100644
--- a/core/include/tee/tee_cryp_pbkdf2.h
+++ b/core/include/tee/tee_cryp_pbkdf2.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2014, Linaro Limited
  */
 
-#ifndef TEE_CRYP_PBKDF2_H
-#define TEE_CRYP_PBKDF2_H
+#ifndef __TEE_TEE_CRYP_PBKDF2_H
+#define __TEE_TEE_CRYP_PBKDF2_H
 
 #include <tee_api_types.h>
 
@@ -13,4 +13,4 @@ TEE_Result tee_cryp_pbkdf2(uint32_t hash_id, const uint8_t *password,
 			   size_t salt_len, uint32_t iteration_count,
 			   uint8_t *derived_key, size_t derived_key_len);
 
-#endif /* TEE_CRYP_PBKDF2_H */
+#endif /* __TEE_TEE_CRYP_PBKDF2_H */
diff --git a/core/include/tee/tee_cryp_utl.h b/core/include/tee/tee_cryp_utl.h
index 98a694a05..7259cc3d6 100644
--- a/core/include/tee/tee_cryp_utl.h
+++ b/core/include/tee/tee_cryp_utl.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2014, Linaro Limited
  */
 
-#ifndef TEE_CRYP_UTL_H
-#define TEE_CRYP_UTL_H
+#ifndef __TEE_TEE_CRYP_UTL_H
+#define __TEE_TEE_CRYP_UTL_H
 
 #include <tee_api_types.h>
 #include <crypto/crypto.h>
diff --git a/core/include/tee/tee_fs.h b/core/include/tee/tee_fs.h
index 1b632d7ad..fce494c6b 100644
--- a/core/include/tee/tee_fs.h
+++ b/core/include/tee/tee_fs.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2014, STMicroelectronics International N.V.
  */
 
-#ifndef TEE_FS_H
-#define TEE_FS_H
+#ifndef __TEE_TEE_FS_H
+#define __TEE_TEE_FS_H
 
 #include <stddef.h>
 #include <stdint.h>
@@ -99,4 +99,4 @@ tee_svc_storage_file_ops(uint32_t storage_id)
 	}
 }
 
-#endif /*TEE_FS_H*/
+#endif /*__TEE_TEE_FS_H*/
diff --git a/core/include/tee/tee_fs_key_manager.h b/core/include/tee/tee_fs_key_manager.h
index ac5763043..736dedf55 100644
--- a/core/include/tee/tee_fs_key_manager.h
+++ b/core/include/tee/tee_fs_key_manager.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2015, Linaro Limited
  */
 
-#ifndef TEE_FS_KEY_MANAGER_H
-#define TEE_FS_KEY_MANAGER_H
+#ifndef __TEE_TEE_FS_KEY_MANAGER_H
+#define __TEE_TEE_FS_KEY_MANAGER_H
 
 #include <tee_api_types.h>
 #include <utee_defines.h>
diff --git a/core/include/tee/tee_fs_rpc.h b/core/include/tee/tee_fs_rpc.h
index ae49682c9..88132952c 100644
--- a/core/include/tee/tee_fs_rpc.h
+++ b/core/include/tee/tee_fs_rpc.h
@@ -7,8 +7,8 @@
  * Interface with tee-supplicant for file operations
  */
 
-#ifndef TEE_FS_RPC_H
-#define TEE_FS_RPC_H
+#ifndef __TEE_TEE_FS_RPC_H
+#define __TEE_TEE_FS_RPC_H
 
 #include <stdbool.h>
 #include <stddef.h>
@@ -46,4 +46,4 @@ TEE_Result tee_fs_rpc_write_final(struct tee_fs_rpc_operation *op);
 TEE_Result tee_fs_rpc_truncate(uint32_t id, int fd, size_t len);
 TEE_Result tee_fs_rpc_remove_dfh(uint32_t id,
 				 const struct tee_fs_dirfile_fileh *dfh);
-#endif /* TEE_FS_RPC_H */
+#endif /* __TEE_TEE_FS_RPC_H */
diff --git a/core/include/tee/tee_obj.h b/core/include/tee/tee_obj.h
index adbe54315..1c23dc065 100644
--- a/core/include/tee/tee_obj.h
+++ b/core/include/tee/tee_obj.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2014, STMicroelectronics International N.V.
  */
 
-#ifndef TEE_OBJ_H
-#define TEE_OBJ_H
+#ifndef __TEE_TEE_OBJ_H
+#define __TEE_TEE_OBJ_H
 
 #include <kernel/tee_ta_manager.h>
 #include <sys/queue.h>
diff --git a/core/include/tee/tee_pobj.h b/core/include/tee/tee_pobj.h
index bd05da4ed..49575ad51 100644
--- a/core/include/tee/tee_pobj.h
+++ b/core/include/tee/tee_pobj.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2014, STMicroelectronics International N.V.
  */
 
-#ifndef TEE_POBJ_H
-#define TEE_POBJ_H
+#ifndef __TEE_TEE_POBJ_H
+#define __TEE_TEE_POBJ_H
 
 #include <stdint.h>
 #include <sys/queue.h>
diff --git a/core/include/tee/tee_supp_plugin_rpc.h b/core/include/tee/tee_supp_plugin_rpc.h
index 2d725c9d0..5e4688a24 100644
--- a/core/include/tee/tee_supp_plugin_rpc.h
+++ b/core/include/tee/tee_supp_plugin_rpc.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2020, Open Mobile Platform LLC
  */
 
-#ifndef TEE_SUPP_PLUGIN_RPC_H
-#define TEE_SUPP_PLUGIN_RPC_H
+#ifndef __TEE_TEE_SUPP_PLUGIN_RPC_H
+#define __TEE_TEE_SUPP_PLUGIN_RPC_H
 
 #include <stdint.h>
 #include <stdbool.h>
@@ -15,4 +15,4 @@ TEE_Result tee_invoke_supp_plugin_rpc(const TEE_UUID *uuid, uint32_t cmd,
 				      void *buf_user, size_t len,
 				      size_t *outlen);
 
-#endif /* TEE_SUPP_PLUGIN_RPC_H */
+#endif /* __TEE_TEE_SUPP_PLUGIN_RPC_H */
diff --git a/core/include/tee/tee_svc.h b/core/include/tee/tee_svc.h
index 27a04c05d..7c1476567 100644
--- a/core/include/tee/tee_svc.h
+++ b/core/include/tee/tee_svc.h
@@ -2,8 +2,8 @@
 /*
  * Copyright (c) 2014, STMicroelectronics International N.V.
  */
-#ifndef TEE_SVC_H
-#define TEE_SVC_H
+#ifndef __TEE_TEE_SVC_H
+#define __TEE_TEE_SVC_H
 
 #include <kernel/ts_manager.h>
 #include <stdint.h>
@@ -74,4 +74,4 @@ TEE_Result syscall_wait(unsigned long timeout);
 TEE_Result syscall_get_time(unsigned long cat, TEE_Time *time);
 TEE_Result syscall_set_ta_time(const TEE_Time *time);
 
-#endif /* TEE_SVC_H */
+#endif /* __TEE_TEE_SVC_H */
diff --git a/core/include/tee/tee_svc_cryp.h b/core/include/tee/tee_svc_cryp.h
index 3932cdf6a..526d6e698 100644
--- a/core/include/tee/tee_svc_cryp.h
+++ b/core/include/tee/tee_svc_cryp.h
@@ -2,8 +2,8 @@
 /*
 * Copyright (c) 2014, STMicroelectronics International N.V.
 */
-#ifndef TEE_SVC_CRYP_H
-#define TEE_SVC_CRYP_H
+#ifndef __TEE_TEE_SVC_CRYP_H
+#define __TEE_TEE_SVC_CRYP_H
 
 #include <tee_api_types.h>
 #include <utee_types.h>
@@ -93,4 +93,4 @@ TEE_Result tee_obj_attr_from_binary(struct tee_obj *o, const void *data,
 				    size_t data_len);
 TEE_Result tee_obj_attr_copy_from(struct tee_obj *o, const struct tee_obj *src);
 
-#endif /* TEE_SVC_CRYP_H */
+#endif /* __TEE_TEE_SVC_CRYP_H */
diff --git a/core/include/tee/tee_svc_storage.h b/core/include/tee/tee_svc_storage.h
index c5d763ade..0cb5fb396 100644
--- a/core/include/tee/tee_svc_storage.h
+++ b/core/include/tee/tee_svc_storage.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2014, STMicroelectronics International N.V.
  */
 
-#ifndef TEE_SVC_STORAGE_H
-#define TEE_SVC_STORAGE_H
+#ifndef __TEE_TEE_SVC_STORAGE_H
+#define __TEE_TEE_SVC_STORAGE_H
 
 #include <tee_api_types.h>
 #include <kernel/tee_ta_manager.h>
@@ -61,4 +61,4 @@ void tee_svc_storage_close_all_enum(struct user_ta_ctx *utc);
 
 void tee_svc_storage_init(void);
 
-#endif /* TEE_SVC_STORAGE_H */
+#endif /* __TEE_TEE_SVC_STORAGE_H */
diff --git a/core/include/tee/tee_ta_enc_manager.h b/core/include/tee/tee_ta_enc_manager.h
index 5e1b2e5c9..e258679d3 100644
--- a/core/include/tee/tee_ta_enc_manager.h
+++ b/core/include/tee/tee_ta_enc_manager.h
@@ -3,8 +3,8 @@
  * Copyright (c) 2019, Linaro Limited
  */
 
-#ifndef TEE_TA_ENC_MANAGER_H
-#define TEE_TA_ENC_MANAGER_H
+#ifndef __TEE_TEE_TA_ENC_MANAGER_H
+#define __TEE_TEE_TA_ENC_MANAGER_H
 
 #include <signed_hdr.h>
 #include <tee_api_types.h>
-- 
2.47.2

